API-Reference/Macros.md

```diff
@@ -0,0 +1,132 @@
+# Macros
+
+## Version Macros
+
+| Macro | Description |
+| ----- | ----------- |
+| `KOKKOS_VERSION` | The Kokkos version; `KOKKOS_VERSION % 100` is the patch level, `KOKKOS_VERSION / 100 % 100` is the minor version, and `KOKKOS_VERSION / 10000` is the major version. |
+
+## General Settings
+
+| Macro | Description |
+| ------| ---------------|
+| `KOKKOS_ENABLE_DEBUG` | Defined if extra debug features are activated. |
+| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled/|
+| `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` | Defined if debug checks for Kokkos::DualView objects are enabled. |
+| `KOKKOS_ENABLE_DEPRECATED_CODE_3` | Defined if features deprecated in major release 3 are still available. |
+| `KOKKOS_ENABLE_DEPRECATION_WARNING` | Defined if deprecated features generate deprecation warnings |
+| `KOKKOS_ENABLE_COMPILER_WARNINGS` | A subset of compiler warnings are enabled for Kokkos. |
```

Comment:
I did not realize we were defining that.  It feels like an oversight.  Can anyone think of a good reason to have it?